### PR TITLE
Частов Вячеслав. Задача 2. Вариант 25. Сортировка пузырьком (алгоритм чет-нечетной перестановки).

### PR DESCRIPTION
**Pull request by ChastovSlava**
Задача заключается в реализации алгоритма пузырьковой сортировки в двух вариантах: последовательном (SEQUENTIAL) и параллельном (MPI), с применением объектно-ориентированного подхода и тестированием функциональности.

Алгоритм пузырьковой сортировки:
1) Последовательно сравнивает соседние элементы массива.
2) Меняет их местами, если порядок нарушен.
3) Повторяет проходы по массиву до полной упорядоченности.

Последовательная версия (SEQUENTIAL):
Выполняет сортировку на одном процессе, используя стандартные структуры данных и алгоритмы.

Параллельная версия (MPI):
1) Использует технологии параллельного программирования (MPI) для распределения данных между процессами.
2) Обеспечивает ускорение за счет одновременной обработки фрагментов массива.

Тестирование:
1) Включает сценарии с различными входными данными: пустой массив, отсортированные данные, обратный порядок, случайные значения и массив с одинаковыми элементами.
2) Проверка корректности работы алгоритмов и соответствия выходных данных ожиданиям.

### CODE REVIEW

**Review by StroganovM — COMMENTED** [review]
Add a test for checking a zero-length array to the mpi version of your program.

**Review by Kvoks — CHANGES_REQUESTED** [review]
Good job, but the seq version of your program does not check for an array filled with real values. Add a test.

**Review by Kvoks — APPROVED** [review]

**Review by StroganovM — APPROVED** [review]

**Review by aobolensk — CHANGES_REQUESTED** [review]

File: tasks/mpi/chastov_v_bubble_sort/perf_tests/main.cpp [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+#include "mpi/chastov_v_bubble_sort/include/ops_mpi.hpp"
+
+TEST(chastov_v_bubble_sort, test_pipeline_run) {
+  boost::mpi::communicator world;
+  int rank = world.rank();
+  std::vector<int> global_vec;
+  std::vector<int> global_out;
+  // Create TaskData
+  std::shared_ptr<ppc::core::TaskData> taskDataPar = std::make_shared<ppc::core::TaskData>();
+  const size_t count_size_vector = 120;
```
[the remark this answers is not in the document]
Comment (ChastovSlava) [reply tasks/mpi/chastov_v_bubble_sort/perf_tests/main.cpp]: fixed

**Review by aobolensk — APPROVED** [review]

**Review by allnes — APPROVED** [review]
